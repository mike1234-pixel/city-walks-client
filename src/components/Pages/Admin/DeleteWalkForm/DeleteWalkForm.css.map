{"version":3,"sourceRoot":"","sources":["DeleteWalkForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"DeleteWalkForm.css"}