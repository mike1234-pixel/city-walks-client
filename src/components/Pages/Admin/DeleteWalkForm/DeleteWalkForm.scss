/** @format */

.delete-walk-form {
  padding-top: 20px;
  padding-bottom: 20px;
}
