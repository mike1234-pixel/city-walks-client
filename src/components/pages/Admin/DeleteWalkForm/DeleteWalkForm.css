.delete-walk-form {
  padding: 20px;
}
