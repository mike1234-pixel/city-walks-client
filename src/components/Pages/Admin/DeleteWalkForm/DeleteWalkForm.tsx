import React, { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios, { AxiosError, AxiosResponse } from "axios"
import toTitleCase from '../../../../functions/toTitleCase'
import SelectedWalk from "../../../../types/PostRequests/SelectedWalk"
import './DeleteWalkForm.scss'

const DeleteWalkForm: React.FC = () => {

  const [walk, setWalk] = useState<string>("")

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "walk":
        setWalk(event.target.value)
        break;
    }
  }

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault()

    let payload: SelectedWalk = {
      walk: toTitleCase(walk)
    };

    axios
      .delete("https://city-walks.herokuapp.com/delete-walk", { data: payload })
      .then((res: AxiosResponse) => {
        if (res.data === "walk deleted") {
          console.log("walk deleted")
        } else {
          console.log("walk not deleted")
        }
      }).catch((err: AxiosError) => {
        console.log(err)
      });;

    alert("Walk Deleted")
    setWalk("")
    window.scrollTo(0, 0);
  }

  return (
    <div>
      <h2>Delete Walk</h2>
      <form onSubmit={handleSubmit} className="delete-walk-form">
        <MDBInput type="text" name="walk" id="walk" value={walk} label="walk name" onChange={handleChange} required />
        <MDBBtn outline color="elegant" type="submit">
          Delete Walk <MDBIcon icon="trash" />
        </MDBBtn>
      </form>
    </div>
  )
}

export default DeleteWalkForm