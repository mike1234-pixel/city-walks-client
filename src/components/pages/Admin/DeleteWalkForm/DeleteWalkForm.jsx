import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios"
import toTitleCase from '../../../../functions/toTitleCase'
import './DeleteWalkForm.css'

const DeleteWalkForm = () => {

    const [walk, setWalk] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "walk":
              setWalk(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            walk: toTitleCase(walk)
          };

        axios
        .delete("http://localhost:5000/delete-walk",{ data: payload } )
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "walk deleted") {
            console.log("walk deleted")
          } else {
            console.log("walk not deleted")
          }
        });

        alert("Walk Deleted")
        setWalk("")
        window.scrollTo(0, 0);
    }

    return (
    <div>
        <h2>Delete Walk</h2>
        <form onSubmit={handleSubmit} className="delete-walk-form">
            <MDBInput type="text" name="walk" id="walk" value={walk} label="walk name" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Delete Walk <MDBIcon icon="trash"/>
          </MDBBtn>
        </form>
    </div>
    )
}

export default DeleteWalkForm