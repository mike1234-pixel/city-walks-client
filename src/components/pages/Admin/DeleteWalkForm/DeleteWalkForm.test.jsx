import renderer from 'react-test-renderer'
import DeleteWalkForm from "./DeleteWalkForm"

// snapshot test
it('DeleteWalkForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <DeleteWalkForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});