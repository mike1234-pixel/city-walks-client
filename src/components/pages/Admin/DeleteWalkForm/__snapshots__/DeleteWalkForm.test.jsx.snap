// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteWalkForm Page renders correctly 1`] = `
<div>
  <h2>
    Delete Walk
  </h2>
  <form
    className="delete-walk-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="walk"
        name="walk"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="walk"
        id=""
        onClick={[Function]}
      >
        walk name
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Delete Walk 
      <i
        className="fa fa-trash"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
